be/src/exec/vectorized/hdfs_scanner_orc.cpp

```diff
@@ -398,39 +411,75 @@ Status HdfsOrcScanner::do_get_next(RuntimeState* runtime_state, ChunkPtr* chunk)
     {
         SCOPED_RAW_TIMER(&_stats.column_read_ns);
         RETURN_IF_ERROR(_orc_adapter->read_next());
-        RETURN_IF_ERROR(_orc_adapter->apply_dict_filter_eval_cache(_orc_row_reader_filter->_dict_filter_eval_cache));
+        RETURN_IF_ERROR(_orc_adapter->apply_dict_filter_eval_cache(_orc_row_reader_filter->_dict_filter_eval_cache,
+                                                                   &_chunk_filter));
     }
     // if there is no rows returned from orc, we just returned.
     // but we have to set num rows to 0, otherwise user may see old data.
     if (_orc_adapter->get_cvb_size() == 0) {
         (*chunk)->set_num_rows(0);
         return Status::OK();
     }
+
     {
+        StatusOr<ChunkPtr> ret;
         SCOPED_RAW_TIMER(&_stats.column_convert_ns);
-        ChunkPtr ptr = _orc_adapter->create_chunk();
-        RETURN_IF_ERROR(_orc_adapter->fill_chunk(&ptr));
-        ChunkPtr result = _orc_adapter->cast_chunk(&ptr);
-        *chunk = std::move(result);
+        if (!_orc_adapter->has_lazy_load_context()) {
+            ret = _orc_adapter->load_chunk();
+        } else {
+            ret = _orc_adapter->load_active_chunk();
+        }
+        RETURN_IF_ERROR(ret);
+        *chunk = std::move(ret.value());
     }
+
     ChunkPtr ck = *chunk;
     // important to add columns before evaluation
     // because ctxs_by_slot maybe refers to some non-existed slot or partition slot.
     _file_read_param.append_not_exised_columns_to_chunk(chunk, ck->num_rows());
     _file_read_param.append_partition_column_to_chunk(chunk, ck->num_rows());
     // do stats before we filter rows which does not match.
     _stats.raw_rows_read += ck->num_rows();
-    for (auto& it : _file_read_param.conjunct_ctxs_by_slot) {
-        // do evaluation.
+
+    size_t chunk_size = ck->num_rows();
+    Filter& filter = _chunk_filter;
+    filter.assign(chunk_size, 1);
+    {
         SCOPED_RAW_TIMER(&_stats.expr_filter_ns);
-        if (_orc_row_reader_filter->is_slot_evaluated(it.first)) {
-            continue;
-        }
-        ExecNode::eval_conjuncts(it.second, ck.get());
-        if (ck->num_rows() == 0) {
-            return Status::OK();
+        for (auto& it : _file_read_param.conjunct_ctxs_by_slot) {
+            // do evaluation.
+            if (_orc_row_reader_filter->is_slot_evaluated(it.first)) {
+                continue;
+            }
+            chunk_size = ExecNode::eval_conjuncts_into_filter(it.second, ck.get(), &filter);
+            if (chunk_size == 0) {
+                break;
+            }
         }
     }
+    if (chunk_size != 0) {
+        (*chunk)->filter(filter);
+    }
+    (*chunk)->set_num_rows(chunk_size);
+    if (!_orc_adapter->has_lazy_load_context()) {
+        return Status::OK();
+    }
+    // if has lazy load fields, skip it if chunk_size ==0
```

Comment:
nit: chunk_size == 0

Reply:
What's the difference? you mean `chunk_size == 0` in comment?

Reply:
yes, just a style problem